core/src/main/java/org/infinispan/interceptors/impl/CacheLoaderInterceptor.java

```diff
@@ -289,11 +290,22 @@ public Object visitSizeCommand(InvocationContext ctx, SizeCommand command) {
    }
 
    private CompletionStage<Long> trySizeOptimization(long flagBitSet) {
-      if (EnumUtil.containsAny(flagBitSet, FlagBitSets.SKIP_CACHE_LOAD | FlagBitSets.SKIP_SIZE_OPTIMIZATION)) {
+      if (EnumUtil.containsAny(flagBitSet, FlagBitSets.SKIP_CACHE_LOAD | FlagBitSets.SKIP_SIZE_OPTIMIZATION)
+            || cacheConfiguration.persistence().passivation()) {
          return NonBlockingStore.SIZE_UNAVAILABLE_FUTURE;
```

Comment:
I would say instead of checking for stores in the CallInterceptor at https://github.com/infinispan/infinispan/pull/10042/files#diff-747afb8da8cb55c279cc22a7209d04ace5633cb9a9f5b95e708e6a848f739964R634 to instead set the SKIP_SIZE_OPTIMIZATION flag on the command here.

Changing it this way means a user that specifies the SKIP_CACHE_LOAD flag that the optimization can work properly.